src/main/resources/cpp-modern-json/struct-source.mustache

```diff
@@ -20,6 +20,10 @@ void {{classname}}::from_json(const ::nlohmann::json& j, {{classname}}& obj) {
 {{/vars}}
 }
 
+bool operator<(const {{classname}}& l, const {{classname}}& r) {
+   return std::tie({{#vars}}l.{{baseName}}{{^-last}},{{/-last}}{{/vars}}) < std::tie({{#vars}}r.{{baseName}}{{^-last}},{{/-last}}{{/vars}});
```

Comment:
[`std::tuple::<`](https://en.cppreference.com/w/cpp/utility/tuple/operator_cmp) will compare the first elements of the tuples first, return the result accordingly if they are unequal, and continue to the second elements of the tuples only if the first elements were equal.

We don't know what order member variables in the model will have, so we don't know which members will be compared first. I guess the semantics don't matter for our use case, but it does feel iffy.

Maybe we can use [unordered set](https://en.cppreference.com/w/cpp/container/unordered_set) instead? This only needs elements to implement equality, not ordering.